File: BankingApp/app/src/main/java/com/example/bankingapp/LoginActivity.java
package com.example.bankingapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;

public class LoginActivity extends AppCompatActivity {

    EditText etUsername, etPassword;
    Button btnLogin, btnSignUp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        etUsername = findViewById(R.id.etUsername);
        etPassword = findViewById(R.id.etPassword);
        btnLogin = findViewById(R.id.btnLogin);
        btnSignUp = findViewById(R.id.btnSignUp);

        btnLogin.setOnClickListener(v -> {
            String username = etUsername.getText().toString().trim();
            String password = etPassword.getText().toString().trim();

            if (username.isEmpty() || password.isEmpty()) {
                Toast.makeText(this, "Please enter both fields", Toast.LENGTH_SHORT).show();
            } else {
                sendLoginRequest(username, password);

            }
        });

        btnSignUp.setOnClickListener(v -> {
            Intent intent = new Intent(LoginActivity.this, SignUpActivity.class);
            startActivityForResult(intent, 1); // Wait for result from SignUpActivity
        });
    }

    private void sendLoginRequest(String username, String password) {
        new Thread(() -> {
            try {
                URL url = new URL("http://192.168.43.186:8089/login");
                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
                conn.setRequestMethod("POST");
                conn.setDoOutput(true);
                conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");

                String data = "username=" + URLEncoder.encode(username, "UTF-8") +
                        "&password=" + URLEncoder.encode(password, "UTF-8");

                OutputStream os = conn.getOutputStream();
                os.write(data.getBytes());
                os.flush();
                os.close();

                int responseCode = conn.getResponseCode();
                InputStream is = (responseCode == 200) ? conn.getInputStream() : conn.getErrorStream();

                BufferedReader reader = new BufferedReader(new InputStreamReader(is));
                StringBuilder response = new StringBuilder();
                String line;
                while ((line = reader.readLine()) != null) {
                    response.append(line);
                }

                runOnUiThread(() -> {
                    if (responseCode == 200 && response.toString().toLowerCase().contains("login successful")){
                        // Login successful – go to OTP verification
                        Intent otpIntent = new Intent(LoginActivity.this, OtpVerificationActivity.class);
                        otpIntent.putExtra("username", username);
                        startActivity(otpIntent);
                        finish();
                    } else {
                        Toast.makeText(this, "Server said: " + response.toString(), Toast.LENGTH_LONG).show();
                    }
                });

            } catch (Exception e) {
                e.printStackTrace();
                runOnUiThread(() ->
                        Toast.makeText(this, "Error: " + e.getMessage(), Toast.LENGTH_LONG).show());
            }
        }).start();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        // After successful sign-up, redirect to homepage
        if (requestCode == 1 && resultCode == RESULT_OK) {
            Intent intent = new Intent(LoginActivity.this, HomeActivity.class);
            startActivity(intent);
            finish();
        }
    }
}

File: OTPReceiverApp/app/src/main/java/com/example/otpreceiverapp/LoginActivity.kt
package com.example.otpreceiverapp

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.net.HttpURLConnection
import java.net.URL
import java.net.URLEncoder

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val usernameField = findViewById<EditText>(R.id.usernameField)
        val passwordField = findViewById<EditText>(R.id.passwordField)
        val loginButtonAppA = findViewById<Button>(R.id.loginButtonAppA1)

        loginButtonAppA.setOnClickListener(View.OnClickListener { v: View? ->
            val username: String = usernameField.getText().toString().trim { it <= ' ' }
            val password: String = passwordField.getText().toString().trim { it <= ' ' }
            if (username.isEmpty() || password.isEmpty()) {
                Toast.makeText(this, "Please enter both fields", Toast.LENGTH_SHORT).show()
            } else {
                performAppALogin(username, password)
            }
        })
    }

    private fun performAppALogin(username: String, password: String) {
        val url = "http://192.168.43.186:8089/loginAppA"
        val requestBody = "username=${URLEncoder.encode(username, "UTF-8")}&password=${URLEncoder.encode(password, "UTF-8")}"

        Thread {
            try {
                val urlConnection = URL(url).openConnection() as HttpURLConnection
                urlConnection.requestMethod = "POST"
                urlConnection.doOutput = true
                urlConnection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded")
                urlConnection.outputStream.use { os ->
                    os.write(requestBody.toByteArray())
                }

                val responseCode = urlConnection.responseCode
                val response = urlConnection.inputStream.bufferedReader().use { it.readText() }

                runOnUiThread {
                    if (responseCode == 200 && response == "LoginAppA success") {
                        Toast.makeText(this, "Login successful", Toast.LENGTH_SHORT).show()
                        goToSpecialCodeScreen(username)
                    } else {
                        Toast.makeText(this, "Login failed: $response", Toast.LENGTH_LONG).show()
                    }
                }
            } catch (e: Exception) {
                e.printStackTrace()
                runOnUiThread {
                    Toast.makeText(this, "Error: ${e.message}", Toast.LENGTH_LONG).show()
                }
            }
        }.start()
    }

    private fun goToSpecialCodeScreen(username: String) {
        val intent = Intent(this, SpecialCodeActivity::class.java)
        intent.putExtra("username", username)
        startActivity(intent)
        finish()
    }
}